srml/kton/src/lib.rs

```diff
@@ -380,85 +382,46 @@ impl<T: Trait> Currency<T::AccountId> for Module<T> {
 
 impl<T: Trait> LockableCurrency<T::AccountId> for Module<T>
 where
-	T::Balance: MaybeSerializeDeserialize,
+	T::Balance: MaybeSerializeDeserialize + Debug,
 {
-	type Moment = T::BlockNumber;
+	type Id = Id<T::Moment>;
 
 	// `amount` > `free_balance` is allowed
-	fn set_lock(
-		id: LockIdentifier,
-		who: &T::AccountId,
-		amount: T::Balance,
-		until: T::BlockNumber,
-		reasons: WithdrawReasons,
-	) {
-		let now = <system::Module<T>>::block_number();
-		let mut new_lock = Some(BalanceLock {
-			id,
-			amount,
-			until,
-			reasons,
-		});
-		let mut locks = Self::locks(who)
-			.into_iter()
-			.filter_map(|l| {
-				if l.id == id {
-					new_lock.take()
-				} else if l.until > now {
-					Some(l)
-				} else {
-					None
-				}
-			})
-			.collect::<Vec<_>>();
-		if let Some(lock) = new_lock {
-			locks.push(lock);
-		}
-		<Locks<T>>::insert(who, locks);
+	fn set_lock(who: &T::AccountId, amount: Self::Balance, id: Self::Id) {
```

Comment:
```Self::Id``` should be able to represent <id, util, reasons>.

Suggest to keep id, reasons in the trait,  and make ```Lock``` generalized to some enum which could represent complex locks such as ```  <util: ...> ```